customer-node/routes/flights.js

```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+const { getFlights } = require('../service/flightsService');
+
+const router = express.Router();
+
+router.get('/', async (req, res, next) => {
+  const requiredFields = ['source', 'destination', 'date'];
+  const missingField = requiredFields.find((field) => !(field in req.query));
+
+  if (missingField) {
+    return res.status(400).json({ message: `${missingField} is required` });
+  }
+
+  const { source, destination, date } = req.query;
+
+  const datere = /([12]\d{3}-(0[1-9]|1[0-2])-(0[1-9]|[12]\d|3[01]))/;
+  if (!date.match(datere)) {
+    return res.status(400).json({ message: 'Invalid Date' });
+  }
+  if (source.length !== 3 || destination.length !== 3) {
+    return res.status(400).json({ message: 'Invalid Airport' });
+  }
+  try {
+    const flights = await getFlights(source, destination, date);
```

Comment:
What if the airportCodes are 3 characters but the airport does not exist?

Reply:
changed from 200 empty array to 400 error